server/src/main/java/org/elasticsearch/common/util/Maps.java

```diff
@@ -81,12 +81,27 @@
      * @param <V>   the type of the values in the map
      * @return an immutable map that contains the items from the specified map with the provided key removed
      */
+    @SuppressWarnings("unchecked")
     public static <K, V> Map<K, V> copyMapWithRemovedEntry(final Map<K, V> map, final K key) {
         Objects.requireNonNull(map);
         Objects.requireNonNull(key);
         assert checkIsImmutableMap(map, key, map.get(key));
-        return map.entrySet().stream().filter(k -> key.equals(k.getKey()) == false)
-            .collect(Collectors.toUnmodifiableMap(Map.Entry::getKey, Map.Entry::getValue));
+        if (map.containsKey(key) == false) {
+            return map;
+        }
```

Comment:
This subtly changes the behavior because someone would assume that the returned value is always a copied map (since it's in the name), and not the original map